### PR TITLE
add Simple_two_layer_model

### PR DESCRIPTION
**Pull request by yazhai**

### CODE REVIEW

File: Simple_two_layer_NN_model/README [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+See Python file for more explanation.
+
+
+
+The example is given in two versions:
+1) A Python script, which was the originally designed from the course, showing the clear walk path of the basic concept of NN.
+
+2) A C++ code based on CUDNN/CUBLAS/CUDA, showing how to use CUDNN to solve the NN problem.
+     It uses the "error_util.h" file coming from the "mnistCUDNN" example composed by nVIDIA.
+     Currtent implemented contents include:
```
Comment (zonca) [inline Simple_two_layer_NN_model/README]: typo

File: Simple_two_layer_NN_model/README [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+          - Layers, its input/output dimension (m,n), the weighted matrix (W[m,n]), a bias vector (b[n])
+          - simple fully connecte forward, using cublasSgemv() function
```
Comment (zonca) [inline Simple_two_layer_NN_model/README]: typo

Comment (zonca) [inline Simple_two_layer_NN_model/README]: can you also add a simple explanation of what the `cublasSgemv` function does?, also for function names on Github it is common to format them with backticks `

File: Simple_two_layer_NN_model/README [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+          - softmax forwards, using cudnnSoftmaxForward()
+          - Activation_TANH forwards, using cudnnActivationForward() with CUDNN_ACTIVATION_TANH
+          - Activation_ReLU forwards, using cudnnActivationForward() with CUDNN_ACTIVATION_RELU
```
Comment (zonca) [inline Simple_two_layer_NN_model/README]: same here, does the function **create** a layer?

**Review by zonca — CHANGES_REQUESTED** [review]
very good, a few comments below,
also please remove the ipynb checkpoint and the *.pyc from the PR.

Can you also add to the `README.md` a few details about how you tested this and how the results of Python compare to the results from CUDA?

If you push to your branch, the PR will be automatically upated.

**Comment by zonca** [comment]
thanks @yazhai ! 

it would be nice to use some more Markdown syntax to make the README file more readable and organized, now it looks like point 2. is not using newlines correctly, see https://github.com/yazhai/mbpol_cuda_devel/blob/1e02105005c1e7ea43565906262127f88adf7aab/Simple_two_layer_NN_model/README.md.

see the guide on Github: https://guides.github.com/features/mastering-markdown/

you can create headings and format equations as code in their own line (with 4 initial spaces).
Now I'll read the README more carefully and send you comments on the content.

File: Simple_two_layer_NN_model/README.md [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+               - Activation_ReLU forwards, using `cudnnActivationForward()` with CUDNN_ACTIVATION_RELU
+     File "two_layer_NN.cu" is the tester, in which layers are created and predictions are made according to the above two layers algorithms.
+     This code works only with single precision float at present.
+
+In the offered example, the Python tester generates a random two layers model, with 4x10 dims in the first layer and 10x3 in the second. 
+Then, it creates 5 random samples (each in a vector of size 4), and is given the correct classifiers. 
+After this step the Python script predicts the scores after all layers, and calculates the losses of all samples according to the classifiers.
+
+In the CUDA/CUDNN tester, the weights/bias/input_samples/classifiers generated randomly by the Python code are copied/pasted into an input file, 
+and are input to the CUDA/CUDNN tester as arraies when the file is loaded.
```
Comment (zonca) [inline Simple_two_layer_NN_model/README.md]: typo: arrays

**Comment by zonca** [comment]
also the Jupyter Notebook is too big, it has a lot of output, it is difficult to read on Github, can you please remove some output?

**Comment by zonca** [comment]
awesome job! once you do those quick fixes to README and ipynb, I can merge this in!

**Comment by yazhai** [comment]
Hello,

The README has been corrected and tested on my side. 
The only problem is, for mathematical expressions, the default markdown does not support Latex style from my browser view.
Meanwhile, the online solutions require external markdown instead of Github's.
So I have to put this part in plain text, but should still be able to read with ease.

Regards,
Yaoguang
2017.07.27

**Comment by zonca** [comment]
yes, now it looks very good!

**Pull request merged**